docs/proposed-api.md

```diff
@@ -0,0 +1,173 @@
+# Proposed API
+
+All connections between VPCs is done via `Peering` object.
+
+2 VPCs can only have a single `Peering` object between them.
+
+External connections are modeled as VPCs where we can separately configure
+how we map incoming traffic to thhe VPC (VNI, VLAN, QinQ, MPLS, etc.)
+
+## Duplicate/Ambiguous routes
+
+Here, there are no duplicate IP restrictions, if there is multipath you just get
+ECMP. We can warn the user.  The policy is based on whatever route we pick.  However,
+there are route metrics to prefer one path to the other.
+
+This helps with the multiple external cases where one VPC is routing to 2 externals
+and we want to use route metrics advertised via BGP to choose routes.
+
+## Questions
+
+Is this implementable?
+
+frostman and mvachhar believe so, others should check
+
+Do we need explict NAT for use cases that don't involve fabric?
+
+Is all round trip routing stateless, or can we specify directional stateful routing?
+
+frostman and mvachhar think that if the expose is not stateless, return routing can
+be based on flow state.  How does this interact with other configuration?
+
+## Use Cases
+
+```yaml
+# Static NAT, VPC 1 -> VPC 2 and vice versa
+# VPC 2 exposes http port 80 on its private subnet 10.2.1.1/32
+# Any IP from VPC 1 can connect to VPC 2 on 10.2.1.1/32
+# All static, no dynamic/stateful
+apiVersion: gateway.githedgehog.com/v1alpha1
+kind: Peering
+metadata:
+  name: vpc-1--vpc-2
+spec:
+  vpc1:
+    ips:
+      - cidr: 10.1.1.0/24
+      # - fromVPCSubnet: subnet1 # just a shorthand for the above
+    as: # Means static Src/Dst NAT for vpc1
+      - 192.168.1.0/24
+    ingress:
+      - allow:
+          stateless: true # it's the only options supported in the first release
+          tcp:
+            dstPort: 443
+  vpc2:
+    ips:
+      - cidr: 10.2.1.0/24
```

Comment:
I'm assuming you mean 10.2.1.1/32 here, according to the comments above

Reply:
Oh, yes, good catch.  Also should be port 80

Reply:
Fixed.